server/src/main/java/org/candlepin/resource/ActivationKeyResource.java

```diff
@@ -161,7 +161,17 @@ public ActivationKeyDTO updateActivationKey(
         @ApiParam(name = "update", required = true) ActivationKeyDTO update) {
 
         ActivationKey toUpdate = this.fetchActivationKey(activationKeyId);
+
         if (update.getName() != null) {
+            String keyName = update.getName()
+                .replace("-", "0").replace("_", "0");
```

Comment:
After talking with @wottop, a better way to approach this rather than do any explicit filtering is to simply add hyphens and underscores to the allowed regex. Could even take the moment to upgrade the regex to better match the format of keys.

Reply:
Yes, I like the idea of pre-compiling the pattern.
Should I move this `AK_CHAR_FILTER` to `org.candlepin.util` package, right now its duplicated to both `ActivationKeyResouce` & `OwnerResource` file.

- Refactored the older code for generating Activation keys.